OrbitLinuxTracing/TracerThread.cpp

```diff
@@ -579,7 +579,14 @@ void TracerThread::ProcessSampleEvent(const perf_event_header& header,
 
   constexpr size_t size_of_uretprobes = sizeof(perf_event_empty_sample);
   bool is_uretprobe = !is_gpu_event && (header.size == size_of_uretprobes);
-  CHECK(is_gpu_event + is_uprobe + is_uretprobe <= 1);
+
+  constexpr size_t size_of_sample = sizeof(perf_event_stack_sample);
+  bool is_sample = !is_gpu_event && (header.size == size_of_sample);
+
+  static_assert(size_of_uprobes != size_of_uretprobes &&
+                size_of_uprobes != size_of_sample &&
+                size_of_uretprobes != size_of_sample);
+  CHECK(is_gpu_event + is_uprobe + is_uretprobe + is_sample <= 1);
```

Comment:
well, apparently I cannot think of a way to replace it with boolean operators so that it readable.

Reply:
It could be worse!